sdks/java/core/src/test/java/org/apache/beam/sdk/io/range/ByteKeyRangeTrackerTest.java

```diff
@@ -151,4 +152,33 @@ public void testSplitAtPosition() {
     assertFalse(tracker.trySplitAtPosition(INITIAL_MIDDLE_KEY));
     assertTrue(tracker.tryReturnRecordAt(true, INITIAL_MIDDLE_KEY));
   }
+
+  /** Tests for {@link ByteKeyRangeTracker#getSplitPointsConsumed()}. */
+  @Test
+  public void testGetSplitPointsConsumed() {
+    ByteKeyRangeTracker tracker = ByteKeyRangeTracker.of(INITIAL_RANGE);
+    assertEquals(0, tracker.getSplitPointsConsumed());
+
+    // Unstarted, should not split. Split points should remain at 0.
+    assertFalse(tracker.trySplitAtPosition(INITIAL_MIDDLE_KEY));
+    assertEquals(0, tracker.getSplitPointsConsumed());
+
+    // Start it, split 1 time, 0 split points consumed
+    assertTrue(tracker.tryReturnRecordAt(true, INITIAL_START_KEY));
+    assertTrue(tracker.trySplitAtPosition(BEFORE_END_KEY));
+    assertEquals(0, tracker.getSplitPointsConsumed());
+
+    // Split 2 times, 1 split point consumed
+    assertTrue(tracker.trySplitAtPosition(INITIAL_MIDDLE_KEY));
```

Comment:
I don't think this logic is right. If you haven't returned any more records, then you haven't consumed any split points because you're still processing the current record.